main.go

```diff
@@ -68,3 +86,50 @@ func getGIF(text string) string {
 	imageIndex := r.Intn(len(s.Data))
 	return s.Data[imageIndex].URL
 }
+
+func resolveEmotion(text string) string {
+	textLower := strings.ToLower(text)
+	if (textLower == "sad" || textLower == "happy") {
+		return text;
+	}
+
+	return fetchEmotion(text)
+}
+
+func fetchEmotion(text string) string {
+	url := "https://apiv2.indico.io/emotion"
+	var jsonStr = []byte(`{"data": "` + text + `", "top_n": 1}`)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
+	req.Header.Set("X-ApiKey", indicoKey)
+	req.Header.Set("client-lib", "golang")
+	req.Header.Set("version-number", "0.9.0")
+	req.Header.Set("Accept", "text/plain")
+
+	resp, err := netClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+
+	defer resp.Body.Close()
+
+	log.Println("response status: ", resp.Status)
+	log.Println("response Headers:", resp.Header)
+	body, _ := ioutil.ReadAll(resp.Body)
+	log.Println("response Body:", string(body))
+
+	var jsonResponse indicoResponse
+	// var jsonResponse map[string]interface{}
+	json.Unmarshal(body, &jsonResponse)
+
+	//var results map[string]interface{} = jsonResponse["results"]
+	results := jsonResponse.Results
+
+	key := text
```

Comment:
what does this do

Reply:
Declaration + default value if results is empty?